#156037248 UI/UX Revamp

codestaintin opened the pull request:
#### What does this PR do?
UI/UX Revamp
#### Description of Task to be completed?
- change list view to grid view
- implement swipe to refresh functionality
- add share float button for share functionality
- implement progress dialog when fetching data from API
#### How should this be manually tested?
- run `git clone` to clone this repository
- `cd` to the repository
- run `git checkout feature/156037248/ui-ux-revamp`
- run app
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
- story title - UI/UX Revamp
- story type - feature
- story id - #156037248
#### Screenshots (if appropriate)
![screenshot_1521729786](https://user-images.githubusercontent.com/4518420/37777502-fef709ce-2de7-11e8-8c2b-8bcf060773c9.png)
![screenshot_1521903152](https://user-images.githubusercontent.com/4518420/37865384-05172c22-2f7c-11e8-89dc-fecfa825fe89.png)
![screenshot_1521903160](https://user-images.githubusercontent.com/4518420/37865385-053970e8-2f7c-11e8-89b1-9a5c034497be.png)


#### Questions:
N/A

[Finishes #156037248]

Commit:
feature(ui-ux-revamp):

- create progress dialog interface in GithubUsersPresenter.java
- implement swipe to refresh functionality in MainActivity.java
- add share float icon to user profile design

[Finishes #156037248]

codestaintin committed:
Merge pull request #24 from codestaintin/feature/156037248/ui-ux-revamp

#156037248 UI/UX Revamp

TheDancerCodes commented:
@codestaintin 

Share functionality works well. Good job. 👏 

A few things for you to fix though:

1. I have noticed that when I switch orientation, your app crashes.
2. Regarding the UI/UX revamp, I would recommend that you improve how it looks following Material Design Guidelines:

   - **Recommendation:** You could implement a Single-line grid list as demonstrated in this link:
https://material.io/guidelines/components/grid-lists.html#grid-lists-specs

3. You also haven't implemented the `CollapsingToolbarLayout` and `Cards` in the DetailActivity.

Commit:
feature(ui-ux-revamp):

- fix configuration change bug
- implement CollapsingToolbarLayout and Cards in the DetailActivity

[Finishes #156037248]

codestaintin commented:
@TheDancerCodes changes have been updated. Thanks for the feedback

TheDancerCodes commented on app/src/main/res/drawable/shadow.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
```
What are you using this xml for?
@codestaintin

codestaintin replied:
This was used to add a shadow to the card image. @TheDancerCodes

codestaintin replied:
And the image appears that way because I used the emulator's screenshot button to take the screenshot, the extra padding it added is the extra edge before the screen on a real phone

@TheDancerCodes

TheDancerCodes replied:
What do you mean by extra edge? Is this being done at runtime? Do you mean that on a real device the image spans the whole width?
@codestaintin

TheDancerCodes commented on app/src/main/res/layout/test_detail.xml:
```diff
@@ -0,0 +1,87 @@
+<?xml version="1.0" encoding="utf-8"?>
```
@codestaintin 
Should this file be in the code seeing that the activity_detail.xml is already present?

TheDancerCodes reviewed (commented):
Kindly look into these comments.
@codestaintin 

Also from your screenshots, it seems like the image is not occupying the whole width of the card as specified by *Single-line grid list*:
the https://material.io/guidelines/components/grid-lists.html#grid-lists-specs
